examples/algorithms/retry.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) NVIDIA
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+// Pull in the reference implementation of P2300:
+#include <execution.hpp>
+
+template <class From, class To>
+using _copy_cvref_t = std::__member_t<From, To>;
+
+template <class From, class To>
+concept _decays_to = std::same_as<std::decay_t<From>, To>;
+
+///////////////////////////////////////////////////////////////////////////////
+// retry algorithm:
+
+// _conv needed so we can emplace construct non-movable types into
+// a std::optional.
+template<std::invocable F>
+  requires std::is_nothrow_move_constructible_v<F>
+struct _conv {
+  F f_;
+  explicit _conv(F f) noexcept : f_((F&&) f) {}
+  operator std::invoke_result_t<F>() && {
+    return ((F&&) f_)();
+  }
+};
+
+// pass through all customizations except set_error, which retries the operation.
+template<class O, class R>
+struct _retry_receiver
+  : std::execution::receiver_adaptor<_retry_receiver<O, R>> {
+  O* o_;
+
+  R&& base() && noexcept { return (R&&) o_->r_; }
+  const R& base() const & noexcept { return o_->r_; }
```

Comment:
Yeah. That looks good. I did confuse base with base class for a bit. If you want to try another name. The function of that receiver is destination or completion or result or output.

Reply:
All the iterator and view adaptors provide accessors named `base()` that return the inner thing.

Reply:
Yeah. It looks weirder when base() is implemented in the derived class.